src/pages/Market/components/AllItem.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useState, useEffect, useCallback } from 'react';
+import SearchIcon from '../../../assets/images/searchicon.png';
+import { Link } from 'react-router-dom';
+import ItemCard from './ItemCard';
+import { getProducts } from '../../../api/itemApi';
+import styles from '../Market.module.css';
+
+const getPageSize = () => {
+    const width = window.innerWidth;
+    if (width < 767) {
+        return 4;
+    } else if (width < 1280) {
+        return 6;
+    } else {
+        return 10;
+    }
+};
+
+interface Item {
+    id: number;
+    name: string;
+    description: string;
+    price: number;
+    tags: string[];
+    images: string[];
+    ownerId: number;
+    ownerNickname: string;
+    favoriteCount: number;
+    createdAt: string;
+}
+
+interface ItemList {
+    totalCount: number;
+    list: Item[];
+}
+
+function AllItem() {
+    const [itemList, setItemList] = useState<ItemList>({ totalCount: 0, list: [] });
+    const [pageSize, setPageSize] = useState(getPageSize());
```

Comment:
P3:
아래처럼 타입을 명시해주시면 number가 아니라 더 정확한 타입으로 좁힐 수 있습니다.
```suggestion
    const [pageSize, setPageSize] = useState<ReturnType<typeof getPageSize>>(getPageSize());
```